host/xtest/pkcs11_1000.c

```diff
@@ -284,6 +284,46 @@ static void xtest_pkcs11_test_1001(ADBG_Case_t *c)
 	}
 
 	Do_ADBG_EndSubCase(c, "Test C_GetMechanism{List|Info}()");
+	Do_ADBG_BeginSubCase(c, "Test C_GetMechanismList() with larger result buffer");
+
+	for (i = 0; i < slot_count; i++) {
+		CK_SLOT_ID slot = slot_ids[i];
+		CK_ULONG real_mecha_count = 0;
+		CK_ULONG alloc_mecha_count = 0;
+		size_t j = 0;
+
+		rv = C_GetMechanismList(slot, NULL, &real_mecha_count);
+		if (!ADBG_EXPECT_CK_OK(c, rv))
+			goto out;
+
+		if (real_mecha_count == 0)
+			continue;
+
+		/* Allocate more memory for mechanisms than required */
+		alloc_mecha_count = real_mecha_count + 16;
+		mecha_count = alloc_mecha_count;
+
+		free(mecha_types);
+		mecha_types = calloc(alloc_mecha_count, sizeof(*mecha_types));
+		if (!ADBG_EXPECT_NOT_NULL(c, mecha_types))
+			goto out;
+
```

Comment:
ditto about `mecha_types[]` that should rather be initialized to 0xFF as 0 is a valid mechanism ID.

Reply:
To match the C_GetAttributeValue test this is now also 0xCC. Please see comment in there.